athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneMetadataHandler.java

```diff
@@ -0,0 +1,249 @@
+import com.amazonaws.athena.connector.lambda.metadata.ListSchemasRequest;
+import com.amazonaws.athena.connector.lambda.metadata.ListSchemasResponse;
+import com.amazonaws.athena.connector.lambda.metadata.ListTablesRequest;
+import com.amazonaws.athena.connector.lambda.metadata.ListTablesResponse;
+
+import com.amazonaws.athena.connector.lambda.metadata.glue.GlueFieldLexer;
+import com.amazonaws.services.glue.AWSGlue;
+
+import com.amazonaws.services.glue.model.GetTablesRequest;
+import com.amazonaws.services.glue.model.GetTablesResult;
+import com.amazonaws.services.glue.model.Table;
+import com.amazonaws.athena.connector.lambda.security.EncryptionKeyFactory;
+import com.amazonaws.services.athena.AmazonAthena;
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+
+import org.apache.arrow.util.VisibleForTesting;
+import org.apache.arrow.vector.complex.reader.FieldReader;
+
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is part of an tutorial that will walk you through how to build a
+ * connector for your custom data source. The README for this module
+ * (athena-example) will guide you through preparing your development
+ * environment, modifying this example Metadatahandler, building, deploying, and
+ * then using your new source in an Athena query.
+ * <p>
+ * More specifically, this class is responsible for providing Athena with
+ * metadata about the schemas (aka databases), tables, and table partitions that
+ * your source contains. Lastly, this class tells Athena how to split up reads
+ * against this source. This gives you control over the level of performance and
+ * parallelism your source can support.
+ * <p>
+ * For more examples, please see the other connectors in this repository (e.g.
+ * athena-cloudwatch, athena-docdb, etc...)
+ */
+public class NeptuneMetadataHandler extends GlueMetadataHandler {
+    private static final Logger logger = LoggerFactory.getLogger(NeptuneMetadataHandler.class);
+
+    // Used to denote the 'type' of this connector for diagnostic purposes.
+    private static final String SOURCE_TYPE = "neptune";
+
+    private NeptuneConnection neptuneConnection;
+    private GraphTraversalSource g;
+    private AWSGlue glue;
+    // private GetTablesRequest glueReq;
+    private static final String glue_database_name = System.getenv("glue_database_name");
+
+    public NeptuneMetadataHandler() {
+        super(false, SOURCE_TYPE);
+        neptuneConnection = new NeptuneConnection(System.getenv("neptune_endpoint"), System.getenv("neptune_port"));
```

Comment:
Do these connections ever go bad? In other connectors (see hbase or docdb) we use a ConnectionFactory to abstract how we get a connection. It makes it easier to test for bad connections and rebuild the connection. As presently written, I'm a bit worried that if this ConnectionObject goes bad that the customer won't ahve any way to recover without either waiting 15 minutes for the Lambda to be restarted or doing a deployment that forces the lambdas to restart.

Reply:
A connection to client is initialized only when query is sent to be processed. 

https://tinkerpop.apache.org/javadocs/current/full/org/apache/tinkerpop/gremlin/driver/Cluster.html

"Note that calling this method does not imply that a connection is made to the server itself at this point. Therefore, if there is only one server specified in the Cluster and that server is not available an error will not be raised at this point. Connections get initialized in the Client when a request is submitted or can be directly initialized via Client.init()."

Reply:
ok, cool. Might be worth adding this as a comment in the code.